pkg/sources/filesystem/filesystem.go

```diff
@@ -223,9 +218,32 @@ func (s *Source) scanFile(ctx context.Context, path string, chunksChan chan *sou
 // filepath or a directory.
 func (s *Source) Enumerate(ctx context.Context, reporter sources.UnitReporter) error {
 	for _, path := range s.paths {
-		item := sources.CommonSourceUnit{ID: path}
-		if err := reporter.UnitOk(ctx, item); err != nil {
-			return err
+		fileInfo, err := os.Stat(filepath.Clean(path))
+		if err != nil {
+			if err := reporter.UnitErr(ctx, err); err != nil {
+				return err
+			}
+			continue
+		}
+		if !fileInfo.IsDir() {
+			item := sources.CommonSourceUnit{ID: path}
+			if err := reporter.UnitOk(ctx, item); err != nil {
+				return err
+			}
+			continue
+		}
+		err = fs.WalkDir(os.DirFS(path), ".", func(relativePath string, d fs.DirEntry, err error) error {
+			if err != nil || d.IsDir() {
```

Comment:
Does this risk throwing away any errors that we were previously capturing?

Reply:
For reference, the [WalkDirFunc](https://pkg.go.dev/io/fs#WalkDirFunc) describes the `err` argument:

> The err argument reports an error related to path, signaling that WalkDir will not walk into that directory. The function can decide how to handle that error; as described earlier, returning the error will cause WalkDir to stop walking the entire tree.

Perhaps there's an argument that we should record the error in the job report, but as it stands we currently ignore any errors given by `WalkDir`: https://github.com/trufflesecurity/trufflehog/blob/8380e1713ea5d3fc21c2f1256ab703f2af30b0de/pkg/sources/filesystem/filesystem.go#L112-L116